Shared/Samples/Show device location with NMEA data sources/ShowDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,296 @@
+        private var tasks = [Task<Void, Never>]()
+        
+        // MARK: Methods
+        
+        deinit {
+            clearTasks()
+        }
+        
+        /// Cancels and removes tasks.
+        private func clearTasks() {
+            tasks.forEach { task in
+                task.cancel()
+            }
+            tasks.removeAll()
+        }
+        
+        /// Gets the first connected and supported Bluetooth accessory with its
+        /// protocol string.
+        /// - Returns: A tuple of the accessory and its protocol,
+        /// or `nil` if no supported accessory exists.
+        func firstSupportedAccessoryWithProtocol() -> (EAAccessory, String)? {
+            for accessory in EAAccessoryManager.shared().connectedAccessories {
+                // The protocol string to establish the EASession.
+                guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                    // Skip the accessories with protocol not for NMEA data transfer.
+                    continue
+                }
+                // Only return the first connected and supported accessory.
+                return (accessory, protocolString)
+            }
+            return nil
+        }
+        
+        /// The Bluetooth accessory picker connected to a supported accessory.
+        func accessoryDidConnect(connectedAccessory: EAAccessory, protocolString: String) {
+            if let dataSource = NMEALocationDataSource(accessory: connectedAccessory, protocol: protocolString) {
+                nmeaLocationDataSource = dataSource
+                start()
+            }
+        }
+        
+        /// Resets the sample, stops the data source, cancels tasks, and resets button states and status strings.
+        func reset() {
+            // Reset buttons states.
+            isResetButtonDisabled = true
+            isSourceMenuDisabled = false
+            
+            // Reset the status text.
+            accuracyStatus = "Accuracy info will be shown here."
+            satelliteStatus = "Satellites info will be shown here."
+            
+            // Reset NMEA location data source.
+            nmeaLocationDataSource = nil
+            autoPanMode = .off
+            
+            Task {
+                // Stop the location display, which in turn stop the data source.
+                await locationDisplay.dataSource.stop()
+            }
+            
+            // Cancel and remove the autopan, location, and satellite observation tasks.
+            clearTasks()
+            
+            // Pause the mock data generation.
+            mockNMEADataSource.stop()
+        }
+        
+        /// Starts the location data source and awaits location and satellite updates.
+        /// - Parameter usingMockedData: Indicates that the location datasource should use mocked data.
+        func start(usingMockedData: Bool = false) {
```

Comment:
This method and `accessoryDidConnect()` should throw and present the error to the user in the case that `try await locationDisplay.dataSource.start()` fails. The actual error message could be presented or a localized error message.

Reply:
Applied.